src/cmd/tools/read_data_files/main/main.go

```diff
@@ -86,6 +87,14 @@ func main() {
 	}
 	log := rawLogger.Sugar()
 
+	filterIds := strings.Split(*idFilter, ",")
+	filterIdRegex, err := regexp.Compile(strings.Join(filterIds, "|"))
+
+	if err != nil {
+		fmt.Printf("Error compiling regular expression from filter %s: %s\n", *idFilter, err)
+		return
```

Comment:
This is a silent failure. The caller of this binary won't be able to tell if anything goes wrong. Should bubble up the failure like `os.Exit(1)` or use `MustCompile()` above.